lock_api/src/remutex.rs

```diff
@@ -62,6 +64,7 @@ impl<R: RawMutex, G: GetThreadId> RawReentrantMutex<R, G> {
                 return false;
             }
             self.owner.store(id, Ordering::Relaxed);
+            // REVIEW: add a debug assert for lock_count == 0?
```

Comment:
Doing so results in a panic. It's not zero here all the time. The failing test is `is_mutex`. I have not yet checked if this is a bug or expected behavior, and I'm giving up for today now.

Reply:
Hm that naively sounds like it would cause the store of 1 below to perhaps overwrite something previous, so that sounds bad?

Reply:
That's just because when we perform the last unlock we don't bother resetting `lock_count` to 0.

Reply:
Can it be reset to zero so a debug assertion can be added? Surely that's not really performance sensitive...

Reply:
Addressed in https://github.com/Amanieu/parking_lot/pull/148